src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -203,15 +283,42 @@ public void PutCounterFromReplication(DocumentsOperationContext context, string
             }
         }
 
-        public string IncrementCounter(DocumentsOperationContext context, string documentId, string name, long value)
+        private void RemoveTombstoneIfExists(DocumentsOperationContext context, string documentId, string name)
+        {
+            using (GetCounterPartialKey(context, documentId, name, out var keyPerfix))
+            {
+                var tombstoneTable = context.Transaction.InnerTransaction.OpenTable(TombstonesSchema, CountersTombstonesSlice);
+
+                if (tombstoneTable.ReadByKey(keyPerfix, out var existingTombstone))
+                {
+                    tombstoneTable.Delete(existingTombstone.Id);
+                }
+            }
+        }
+
+        private readonly HashSet<string> _tableCreated = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
It's it. It was meant to prevent from calling it on every operation that modifies a counter. Actually I borrowed this code from revisions storage - should it be removed from both?

Reply:
It's probably cheaper not to.

Reply:
I'll stay with current code then.